File: 백준/Silver/7568. 덩치/덩치.py
import sys

input = sys.stdin.readline

N = int(input())
A = list(list(map(int,input().split())) for _ in range(N))
ans = [1] * N

for i in range(len(A)) :
    for j in range(len(A)) :
        if i == j : continue
        if A[i][0] < A[j][0] and A[i][1] < A[j][1] :
            ans[i] += 1

print(*ans)

File: 백준/Silver/7568. 덩치/덩치.js
const readLine = require('fs').readFileSync('dev/stdin').toString().trim();
const [N, ...peoples] = readLine.split('\n');
const answer = new Array(Number(N)).fill(1);

for (let i = 0; i < peoples.length; i++) {
  for (let j = 0; j < peoples.length; j++) {
    if (i === j) continue;
    const [weightA, heightA] = peoples[i].split(' ').map(Number);
    const [weightB, heightB] = peoples[j].split(' ').map(Number);
    if (weightA < weightB && heightA < heightB) {
      answer[i]++;
    }
  }
}

console.log(answer.join(' '));